src/components/Form/FormSection.tsx

```diff
@@ -24,140 +24,154 @@ export interface FormSectionProps<TData extends GenericFormValues, TErrors> {
   saveButton?: string;
   cancelButton?: string;
   canSave?: (values: TData) => boolean;
-  submit?: (values: TData) => Promise<{ errors?: string[] }> | void;
-  children?: never;
+  onSubmit?: (values: TData) => Promise<{ errors?: string[] }> | void;
   content: (args: FormSectionContentArgs<TData, TErrors>) => ReactNode;
+  children?: never;
 }
 
-export function FormSection<TData extends GenericFormValues>(props: FormSectionProps<TData, string[]>): ReactElement<FormSectionProps<TData, string[]>> | null {
-  const sectionId = useOpaqueId(props.label);
+export function FormSection<TData extends GenericFormValues>({
+  className,
+  panelHeaderClassName,
+  label,
+  editButton,
+  saveButton,
+  cancelButton,
+  canSave,
+  onSubmit,
+  content,
+}: FormSectionProps<TData, string[]>): ReactElement<FormSectionProps<TData, string[]>> | null {
+  const sectionId = useOpaqueId(label);
   const {
     formState: { values, activeSectionId, isDirty, errors: formErrors, isBusy },
     setFormState,
     isLoading,
     cancel,
   } = useContext(FormContext) as FormContext<TData>;
 
-  function onChange(event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) {
-    if (!event.currentTarget) return;
+  const isEditing = sectionId === activeSectionId;
 
-    const name = event.currentTarget.name;
-    const value = event.currentTarget.type === 'checkbox' ? (event.currentTarget as HTMLInputElement).checked : event.currentTarget.value;
+  const onChange = useCallback(
+    function onChange(event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>) {
+      if (!event.currentTarget) return;
 
-    if (activeSectionId !== sectionId) {
-      cancel();
-    }
+      const name = event.currentTarget.name;
+      const value = event.currentTarget.type === 'checkbox' ? (event.currentTarget as HTMLInputElement).checked : event.currentTarget.value;
```

Comment:
I have no real preference for one of these snippets over the other, though I tend to try to avoid intermediate variables. However, I refactored to use destructuring more efficiently.